clang/lib/CIR/Dialect/IR/CIRDataLayout.cpp

```diff
@@ -112,6 +113,8 @@ class StructLayoutMap {
 
 CIRDataLayout::CIRDataLayout(mlir::ModuleOp modOp) : layout{modOp} {
   reset(modOp.getDataLayoutSpec());
+  typeSizeInfo = mlir::cast<TypeSizeInfoAttr>(
+      modOp->getAttr(cir::CIRDialect::getTypeSizeInfoAttrName()));
```

Comment:
It's best we don't force these attributes to be always around for any given tests, too taxing, I believe we can use another way to look at this problem, sorry that I didn't get this the first time and you had to change the tests. Instead, if no `typeSizeInfo` is available, create one here; either hardcode or query the `layout` for good proxies. Suggestions: for size_t use the pointer type size and the other ones can be just hardcoded to 8 and 32.
```
#cir.type_size_info<
  char = 8,
   int = 32, size_t = 64>}
```